web_services/foxx/brightid/db.js

```diff
@@ -319,168 +166,145 @@ function invite(inviter, invitee, groupId, data, timestamp) {
-  if (! groupsColl.exists(groupId)) {
-    throw new errors.GroupNotFoundError(groupId);
-  }
-  const group = groupsColl.document(groupId);
+  const group = getGroup(groupId);
   if (! group.admins || ! group.admins.includes(dismisser)) {
     throw new errors.NotAdminError();
   }
   deleteMembership(groupId, dismissee, timestamp);
 }
 
-function loadUser(id) {
-  return query`RETURN DOCUMENT(${usersColl}, ${id})`.toArray()[0];
-}
-
-function userScore(key) {
-  return query`
-    FOR u in ${usersColl}
-      FILTER u._key  == ${key}
-      RETURN u.score
-  `.toArray()[0];
+function getUser(id) {
+  if (! usersColl.exists(id)) {
+    throw new errors.UserNotFoundError(id);
+  }
+  return usersColl.document(id);
 }
+const checkUserExists = getUser;
 
 function createUser(key, timestamp) {
-  // already exists?
-  const user = loadUser(key);
-
-  if (!user) {
+  if (!usersColl.exists(key)) {
     return usersColl.insert({
-      score: 0,
       signingKeys: [urlSafeB64ToB64(key)],
       createdAt: timestamp,
       _key: key
     });
   } else {
-    return user;
+    return usersColl.document(key);
   }
 }
 
-function hasPrimaryGroup(key) {
-  const groupIds = usersInGroupsColl.byExample({
-    _from: 'users/' + key
-  }).toArray().map(ug => ug._to.replace('groups/', ''));
-  const groups = groupsColl.documents(groupIds).documents;
-  return groups.filter(group => group.type == 'primary').length > 0;
+function familyGroup(key) {
+  return userMemberships(key).map(
+    m => groupsColl.document(m.id)
+  ).find(g => g.type == 'family');
 }
 
-function createGroup(groupId, key1, key2, inviteData2, key3, inviteData3, url, type, timestamp) {
-  if (! ['general', 'primary'].includes(type)) {
-    throw new errors.InvalidGroupTypeError(type);
-  }
-
+function createGroup(groupId, key, url, type, timestamp) {
   if (groupsColl.exists(groupId)) {
     throw new errors.DuplicateGroupError();
   }
 
-  const conns = connectionsColl.byExample({
-    _to: 'users/' + key1
-  }).toArray().map(u => u._from.replace("users/", ""));
-  if (conns.indexOf(key2) < 0 || conns.indexOf(key3) < 0) {
-    throw new errors.InvalidCoFoundersError();
-  }
-
-  const founders = [key1, key2, key3].sort()
-  if (type == 'primary' && founders.some(hasPrimaryGroup)) {
-    throw new errors.AlreadyHasPrimaryGroupError();
-  }
-
-  groupsColl.insert({
+  const group = {
     _key: groupId,
-    score: 0,
-    isNew: true,
-    admins: founders,
+    admins: [key],
     url,
     type,
     timestamp,
-    founders
-  });
+  }
 
-  // Add the creator and invite other cofounders to the group now.
-  // The other two "co-founders" have to join using /membership
-  addUserToGroup(groupId, key1, timestamp);
-  invite(key1, key2, groupId, inviteData2, timestamp);
-  invite(key1, key3, groupId, inviteData3, timestamp);
+  if (type == 'family') {
+    const fg = familyGroup(key);
+    if (fg && fg.head == key) {
+      throw new errors.AlreadyIsFamilyHead();
+    }
```

Comment:
It's fine to be the head of multiple family groups. A family group where you are the head is not considered your family group.

Reply:
Done